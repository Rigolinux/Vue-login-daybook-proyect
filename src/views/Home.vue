<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <button class="btn btn-danger" @click="gotoDaybook"> add</button>
    <button class="btn btn-primary" @click="gotoDaybook" > add</button>
    <button class="btn btn-secondary" @click="gotoDaybook" > add</button>
  </div>
</template>

<script>
// @ is an alias to /src

export default {
  name: 'Home',
  components: {
    
  },
  methods: {
    gotoDaybook() {
      this.$router.push({ name: 'no-entry' })
    }
  }

}
</script>
