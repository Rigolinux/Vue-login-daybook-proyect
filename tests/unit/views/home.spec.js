import { shallowMount } from "@vue/test-utils"
import Home from "@/views/Home.vue"
describe('HOME TEST', () => {

    test('Debe de hacer match con el snapshot', () => {
        const wrapper = shallowMount(Home)
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('hacer click en un boton debe redireccionarn a no entry',()=>{

        // testing with router
        const mockRouter ={
            push: jest.fn()
        }
        const wrapper = shallowMount(Home,{
            global:{
                mocks:{
                    $router: mockRouter
                } // aqui se puede pasar el mockRouter
            }
        })

        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({name:'no-entry'})
    })
})