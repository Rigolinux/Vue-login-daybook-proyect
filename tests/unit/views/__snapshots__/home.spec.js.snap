// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOME TEST Debe de hacer match con el snapshot 1`] = `<div class="home"><img alt="Vue logo" src=""><button class="btn btn-danger"> add</button><button class="btn btn-primary"> add</button><button class="btn btn-secondary"> add</button></div>`;
